Lib/graphlib.py

```diff
@@ -321,6 +322,7 @@ def as_transitive(graph):
                 seen.add(child)
                 continue
 
+            node = path[-1]
```

Comment:
This line fixes a serious bug that the tests didn't catch. I need to get a test around this.

Reply:
Added a failing test